test/e2e/e2e.go

```diff
@@ -57,126 +49,11 @@ import (
-	}
-
-	// Ensure all pods are running and ready before starting tests (otherwise,
-	// cluster infrastructure pods that are being pulled or started can block
-	// test pods from running, and tests that ensure all pods are running and
-	// ready will fail).
-	podStartupTimeout := framework.TestContext.SystemPodsStartupTimeout
-	// TODO: In large clusters, we often observe a non-starting pods due to
-	// #41007. To avoid those pods preventing the whole test runs (and just
-	// wasting the whole run), we allow for some not-ready pods (with the
-	// number equal to the number of allowed not-ready nodes).
-	if err := e2epod.WaitForPodsRunningReady(c, metav1.NamespaceSystem, int32(framework.TestContext.MinStartupPods), int32(framework.TestContext.AllowedNotReadyNodes), podStartupTimeout, map[string]string{}); err != nil {
-		framework.DumpAllNamespaceInfo(c, metav1.NamespaceSystem)
-		framework.LogFailedContainers(c, metav1.NamespaceSystem, e2elog.Logf)
-		runKubernetesServiceTestContainer(c, metav1.NamespaceDefault)
-		e2elog.Failf("Error waiting for all pods to be running and ready: %v", err)
-	}
-
-	if err := framework.WaitForDaemonSets(c, metav1.NamespaceSystem, int32(framework.TestContext.AllowedNotReadyNodes), framework.TestContext.SystemDaemonsetStartupTimeout); err != nil {
-		e2elog.Logf("WARNING: Waiting for all daemonsets to be ready failed: %v", err)
-	}
-
-	// Log the version of the server and this client.
-	e2elog.Logf("e2e test version: %s", version.Get().GitVersion)
-
-	dc := c.DiscoveryClient
-
-	serverVersion, serverErr := dc.ServerVersion()
-	if serverErr != nil {
-		e2elog.Logf("Unexpected server error retrieving version: %v", serverErr)
-	}
-	if serverVersion != nil {
-		e2elog.Logf("kube-apiserver version: %s", serverVersion.GitVersion)
-	}
-
-	// Obtain the default IP family of the cluster
-	// Some e2e test are designed to work on IPv4 only, this global variable
-	// allows to adapt those tests to work on both IPv4 and IPv6
-	// TODO(dual-stack): dual stack clusters should pass full e2e testing at least with the primary IP family
-	// the dual stack clusters can be ipv4-ipv6 or ipv6-ipv4, order matters,
-	// and services use the primary IP family by default
-	// If we´ll need to provide additional context for dual-stack, we can detect it
-	// because pods have two addresses (one per family)
-	framework.TestContext.IPFamily = getDefaultClusterIPFamily(c)
-	e2elog.Logf("Cluster IP family: %s", framework.TestContext.IPFamily)
-
-	// Reference common test to make the import valid.
-	commontest.CurrentSuite = commontest.E2E
-
-	if framework.TestContext.NodeKiller.Enabled {
-		nodeKiller := framework.NewNodeKiller(framework.TestContext.NodeKiller, c, framework.TestContext.Provider)
-		nodeKillerStopCh = make(chan struct{})
-		go nodeKiller.Run(nodeKillerStopCh)
-	}
-	return nil
-
-}, func(data []byte) {
+var _ = ginkgo.SynchronizedBeforeSuite(framework.SetupSuite, func(data []byte) {
```

Comment:
Continuing my other comment: it would be cleaner to make `framework.SetupSuite` a normal function and let `e2e.go` deal with the `ginkgo.SynchronizedBeforeSuite` API.

This code here then would be:
```
var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
    framework.SetupSuite()
    return nil
}, func(data []byte) {
})
```

Reply:
This is why the SetupSuite had a `[]byte` return value, but the way you suggested looks cleaner. I'll change it.